### PR TITLE
Merge `test-branch` to `development`  branch

### PR DESCRIPTION
**Pull request by o-rashad**
- Implement `Jest` node package to test JavaScript `classes`, `modules` and functions.

### CODE REVIEW

File: __tests__/models/project.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const actual = new Project('Homeworks');
+
+    expect(actual.name).toBe(expected);
+  });
+
+  it('adds a todo object to the project object todos variable', () => {
+    const todo = new Todo('Todo');
+    const actual = new Project('Homeworks');
+
+    expect(actual.addTodo(todo)).toBe(1);
```
Comment (whiz25) [inline __tests__/models/project.test.js]: Add tests for `saveToLocal`, `deleteProject`, and `displayProjects` functions. All functions have to be tested.

File: __tests__/models/todo.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Todo from '../../src/todo';
+
+const titleInput = 'todoTitle';
+const todo = new Todo(titleInput);
+
+describe('Todo Class tests suite', () => {
```
Comment (whiz25) [inline __tests__/models/todo.test.js:9]: Add tests for `showTodos`, `saveTodo`, `deleteTodo` and other functions. You may need to restructure your code for easier testing. You mixed is logic with Dom logic in common.js. Am guessing this is why you could not test those functions. 
The best thing to do is to move those functions to the to-do.js module so you can test easily.

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes required 🙏🏿 🙏🏿 🙏🏿 
Nice work on the testing. You did well. 
### changes 🔕 🔕 🔕 
- check comments for suggestions

### Martin Afani
#### Slack: whiz25

File: __tests__/models/todo.test.js (hunk 2 above)
Comment (o-rashad) [reply __tests__/models/todo.test.js:9]: I have done testing all the testable functions available 
however I couldn't find those functions `saveTodo`, `deleteTodo`

Please make sure those exist in my project. 😃

File: src/project.js [hunk 3]
```diff
@@ -4,9 +4,17 @@ export default class Project {
   constructor(name) {
     this.name = name;
     this.todos = [];
+  }
+
+  static saveToLocal(project) {
```
Comment (rindrajosia) [inline src/project.js:7]: - [ ] [REQUIRED] All functions should be tested

File: src/project.js [hunk 4]
```diff
@@ -40,16 +52,7 @@ export default class Project {
...
-        Project.deleteProject(index);
-        Project.displayProjects(JSON.parse(localStorage.projects));
-        Project.projectsEvents(JSON.parse(localStorage.projects));
-        common.showTodos(JSON.parse(localStorage.projects)[index]);
-      });
-    });
+    return projectsList;
   }
 
   static projectsEvents = () => {
```
Comment (rindrajosia) [inline src/project.js]: - [ ] All functions should be tested

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi <STUDENT_MENTION>, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [ ] All functions should be tested:
You can use the --coverage flag to get information:

As you can see in this screenshot for example, 50% of functions inside project.js are tested,...

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] Always remember to add false test cases to your project.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by mke2111 — APPROVED** [review]
### Hi @od-c0d3r, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉** 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------

**Comment by o-rashad** [comment]
Thank you @whiz25 , @rindrajosia , @mke2111 for the reviews ☺️👍🏼

**Pull request merged**